packages/node-core/src/indexer/storeCache/cacheModel.ts

```diff
@@ -45,6 +54,14 @@ export class CachedModel<
       getCacheOptions.max = this.config.storeGetCacheSize;
     }
     this.getCache = new GetData<T>(getCacheOptions);
+    if (Object.keys(this.model.associations).length > 0) {
+      this.hasAssociations = true;
+    }
+  }
+
+  init(storeFlushInOrder = false, getNextStoreOperationIndex: () => number) {
```

Comment:
I think we should always keep the operation index even if we don't flush in order. It will simplify the code and doesn't have much overhead.